Mage.Sets/src/mage/cards/s/ShareTheSpoils.java

```diff
@@ -0,0 +1,316 @@
+                "exile the top card of each player's library.";
+    }
+}
+
+class ShareTheSpoilsExileCardFromEveryoneEffect extends OneShotEffect {
+
+    public ShareTheSpoilsExileCardFromEveryoneEffect() {
+        super(Outcome.Exile);
+    }
+
+    public ShareTheSpoilsExileCardFromEveryoneEffect(final ShareTheSpoilsExileCardFromEveryoneEffect effect) { super(effect); }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (source == null) { return false; }
+
+        PlayerList players = game.getState().getPlayersInRange(source.getControllerId(), game);
+        for (UUID playerId : players) {
+            Player player = game.getPlayer(playerId);
+            if (player == null) { continue; }
+
+            Card topLibraryCard = player.getLibrary().getFromTop(game);
+            if (topLibraryCard == null) { continue; }
+
+            player.moveCardsToExile(
+                    topLibraryCard,
+                    source,
+                    game,
+                    true,
+                    CardUtil.getExileZoneId(game, source),
+                    CardUtil.getSourceName(game, source)
+            );
+        }
+        return true;
+    }
+
+    @Override
+    public ShareTheSpoilsExileCardFromEveryoneEffect copy() { return new ShareTheSpoilsExileCardFromEveryoneEffect(this); }
+}
+
+//-- Play a card Exiled by Share the Spoils --//
+class ShareTheSpoilsPlayExiledCardEffect extends AsThoughEffectImpl {
+
+    ShareTheSpoilsPlayExiledCardEffect() {
+        super(AsThoughEffectType.PLAY_FROM_NOT_OWN_HAND_ZONE, Duration.WhileOnBattlefield, Outcome.PutCardInPlay);
+        staticText = "During each player's turn, " +
+                "that player may play a land or cast a spell from among cards exiled with {this}, " +
+                "and they may spend mana as though it were mana of any color to cast that spell";
+    }
+
+    private ShareTheSpoilsPlayExiledCardEffect(final ShareTheSpoilsPlayExiledCardEffect effect) { super(effect); }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        Card card = game.getCard(sourceId);
+        if (card == null) { return false; }
+
+        // Have to play on your turn
+        if (!game.getActivePlayerId().equals(affectedControllerId)) { return false; }
+
+        // Not in exile
+        if (game.getState().getZone(card.getMainCard().getId()) != Zone.EXILED) { return false; }
+
+        // TODO: This is a workaround for #8706, remove when that's fixed.
+        int zoneChangeCounter = game.getState().getZoneChangeCounter(source.getSourceId());
+        // Not a card exiled with this Share the Spoils
+        ExileZone exileZone = game.getExile().getExileZone(CardUtil.getExileZoneId(game, source.getSourceId(), zoneChangeCounter));
+        if (exileZone == null) { return false; }
+        if (!exileZone.contains(sourceId)) { return false; }
```

Comment:
Must use auto-format for all new and non GUI files. Example for IntelliJ IDEA: <CTRL+ALT+L>.